Migracao refatoracao modelos 202312

robertatakenaka opened the pull request:
#### O que esse PR faz?
Simplifica os modelos que participam do módulo de migração e publicação e, futuramente, também de ingresso.

#### Onde a revisão poderia começar?
Por commits

#### Como este poderia ser testado manualmente?
Preparação
- Tenha core executando (local.yml adaptado para não coincidir as portas)
- Tenha opac_5 executando (usar o yml padrão). Siga as 2 primeiras instruções de https://github.com/scieloorg/opac_5/pull/55
- Tenha minio executando https://min.io/docs/minio/container/index.html
- Acesse a app minio e registre a chave de acesso, este dado será usado no arquivo bigbang
- Faça uma cópia do arquivo `bigbang/.envs/.bigbang.example.for.dev.purpose.only` e edite colocando as configurações adequadas e observe se o JSON está bem formado.

Executando:

```console
make django_bash
```

```console
python manage.py runscript start --script-args adm bigbang/.envs/.bigbang false
```
Sendo **bigbang/.envs/.bigbang**, o nome do arquivo copiado e editado e **false** indica que as tasks envolvidas não estarão habilitadas para executar automaticamente, pois para teste, idealmente é executar cada uma por vez para avaliar o resultado.
Acesse a área administrativa de tasks e execute cada uma na seguinte sequência:

1. migrate_title_databases (a)
2. migrate_issue_databases (a)
3. create_or_update_journals
4. create_or_update_issues
5. task_migrate_document_records (b)
6. task_migrate_document_files (b)
7. publish_journals
8. publish_issues
9. get_xmls
10. generate_sps_packages
11. create_or_update_articles
12. publish_articles

(a) podem ser executados simultaneamente
(b) podem ser executados simultaneamente
Antes de executar 9, espere o suficiente para terminar 5 e 6

As tarefas 1, 2, 5, respectivamente vão criar:
- MigratedJournal e JournalProc
- MigratedIssue e IssueProc
- MigratedArticle e ArticleProc e HTMLXML
Consulte as áreas administrativas Migration e Processing

As tarefas 3, 4, 11 atualizam estes mesmos modelos e criam Journal, Issue, Article
A tarefa 6, coleta os arquivos do site clássico das pastas bases, bases-work, htdocs, serial, os arquivos pdf, xml, html, imagens. O resultado pode ser visto em Migration > Migrated files

As tarefas 7, 8, 12, publicam os dados no opac

A tarefa 9 gera o XML a partir do HTML ou recupera o XML original coletado no passo 6.
A tarefa 10, remonta o pacote XML e, a partir deste ponto, os fluxos de migração e de ingresso terão um caminho em comum. Consultar Processing > SPS Pkg

Apesar de as tarefas estarem agendadas, elas somente executarão se os status estão de acordo com as tarefas.

#### Algum cenário de contexto que queira dar?
(TODO)
No lugar do script, em produção, crie uma task com "start" cujos parâmetros devem ser:
```json
'{"username": username, "activate_subtasks": activate_subtasks, "config": config}'
```
sendo:
- username o nome de um usuário registrado (específico para as tarefas de migração / publicação)
- activate_subtasks (true ou false), sendo true para deixar as tarefas habilitadas
- config: (json) conteúdo similar a `bigbang/.envs/.bigbang.example.for.dev.purpose.only`


### Screenshots
n/a

#### Quais são tickets relevantes?
n/a

### Referências
n/a

robertatakenaka committed:
Reduz o modelo de Journal e remove SciELOJournal

robertatakenaka committed:
Remove SciELOIssue e faz melhorias em Issue

robertatakenaka committed:
Simplifica article.models

robertatakenaka committed:
Remove as migrações de bases de dados, por haver conflitos com as mudanças atuais (diretórios de FileField excluídos, sem uso)

robertatakenaka committed:
Adiciona as novas migrações

robertatakenaka committed:
Adiciona importação de Q

robertatakenaka committed:
Corrige a expressão para contagem de valores None

robertatakenaka committed:
Corrige ausência de importações e nome de parâmetro

robertatakenaka committed:
Remove variáveis não definidas e cria definição ausentes

robertatakenaka committed:
Remove article.tasks

robertatakenaka committed:
Adiciona article.exceptions e journal.exceptions

robertatakenaka committed:
Adiciona article.controller

robertatakenaka committed:
Adiciona article, issue, journal wagtail_hooks

robertatakenaka committed:
Adequa os módulos da app upload devido às mudanças

robertatakenaka committed:
Cria bigbang.utils.scheduler para criar tarefas

robertatakenaka committed:
Cria as tarefas específicas para a migração e publicação

robertatakenaka committed:
Cria função e arquivo modelo de configuração inicial

robertatakenaka committed:
Cria a tarefa que inicializa o sistema

robertatakenaka committed:
Cria o script para fazer a chamada da tarefa que inicia o sistema

robertatakenaka committed:
Adiciona local.yml personalizado para executar com opac_5 e core

robertatakenaka committed:
Adiciona módulo para fazer requisições Web

robertatakenaka committed:
Adiciona bigbang em INSTALLED_APPS

robertatakenaka committed:
Move fixtures do site clássico da app collection para migration e adiciona novos fixtures

robertatakenaka committed:
HTMLXML - adiciona mais detalhes sobre os artigos

- Adiciona HTMLXML.pdf_langs
- Aumenta max_length de HTMLXML.html_translation_langs
- Atribui valor padrão para document.xml_body_and_back
- Corrige a obtenção dos idiomas de html translation

samuelveigarangel commented on journal/models.py:
```diff
@@ -54,10 +61,10 @@ class Meta:
 
     def __str__(self):
-        return "%s - %s" % (self.issnl, self.title) or ""
+        return self.title
```
Aqui não poderá ocorrer um erro se title for nulo?
"TypeError: __str__ returned non-string (type NoneType)"

robertatakenaka committed:
Adiciona pdf_langs para HTMLXML em wagtail_hooks (migration)

robertatakenaka committed:
Faz correções em package.models
- Corrige a validação de components registrados no MinIO (a contagem anterior não considerava que com a otimização novos componentes são criados)
- Corrige a geração do HTML a partir do XML
- Corrige a validação dos textos / idiomas

robertatakenaka committed:
Corrige a ausência de chamada de  PidProviderXML.save()

robertatakenaka committed:
Remove Event.abstract = True

robertatakenaka committed:
Remove o atributo "error" de  proc.wagtail_hooks

robertatakenaka committed:
Refatora proc.models, agregando o resultado da operação em Operation

robertatakenaka committed:
Adiciona main_lang na chamada de article_proc.update()

robertatakenaka committed:
Adiciona campos para serem apresentados em panels

robertatakenaka committed:
Volta atributo Journal.short_title removido por engango

robertatakenaka committed:
Adiciona tratamento de ausência de logo do periódico e marca FIXME para resolver futuramente

robertatakenaka committed:
Melhoria : setup aceita configuração no formato JSON no lugar de arquivo de configuração

robertatakenaka committed:
Corrige OfficialJournal.__str__, garantindo retorno de str()

robertatakenaka committed:
Remove chaves de segunça falsas para evitar mensagem de erro de GitGuardian

robertatakenaka committed:
Adiciona as migrações de banco de dados
